lib/collection/src/collection_manager/holders/segment_holder.rs

```diff
@@ -123,7 +123,7 @@ impl Clone for LockedSegment {
     }
 
     fn clone_from(&mut self, source: &Self) {
-        *self = source.clone();
+        self.clone_from(source);
```

Comment:
This is a recursive call 🙈

Reply:
I was too quick there :face_with_head_bandage: